packages/ragbits-document-search/tests/cli/test_ingest.py

```diff
@@ -0,0 +1,47 @@
+import asyncio
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+from typer.testing import CliRunner
+
+from ragbits.document_search.cli import ds_app, state
+
+projects_dir = Path(__file__).parent.parent / "unit" / "testprojects"
+
+# So that we can import the factory functions
+sys.path.append(str(projects_dir))
+
+# Path to the factory function that creates the test document search instance
+factory_path = "project_with_instance_factory.factories:create_document_search_instance_223"
+
+
+@pytest.mark.parametrize(
+    ("pattern", "num_expected"),
+    [
+        ("*", 4),
+        ("test?.txt", 3),
+        ("test??.txt", 1),
+        ("test*.txt", 4),
+    ],
+)
+def test_ingest(pattern: str, num_expected: int) -> None:
+    runner = CliRunner(mix_stderr=False)
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
nit: you should be able to use pytest's built-in [`tmp_path` fixture](https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmp-path-fixture)